neurokit2/misc/random.py

```diff
@@ -0,0 +1,57 @@
+import copy
+import numbers
+
+import numpy as np
+
+
+def check_rng(seed=None):
+    # If seed is an integer, use the legacy RandomState generator, which has better compatibililty
+    # guarantees but worse statistical "randomness" properties and higher computational cost
+    # See: https://numpy.org/doc/stable/reference/random/legacy.html
+    if isinstance(seed, numbers.Integral):
+        return np.random.RandomState(seed)
+    # If seed is already a random number generator return it as it is
+    if isinstance(seed, (np.random.Generator, np.random.RandomState)):
+        return seed
+    # If seed is something else, use the new Generator class
+    # Note: to initialise the new generator class with an integer seed, use, e.g.:
+    # check_rng(np.random.SeedSequence(123))
+    return np.random.default_rng(seed)
+
+
+def spawn_rng(rng, n_children=1):
```

Comment:
This could probably be made an internal `_spawn_rng` and not exported as it's only used in get_children_rng afaik

Reply:
It is only used internally for now but I would leave it as public as spawning an rng is good practice whenever several RNGs are needed. For example, if one needs to simulate several signals (e.g. several patients in a study) one should not reuse the same seed or use seed, seed+1, seed+2... but rather 
```python
master_seed = 42
patients_RNGs = nk.misc.spawn_rng(master_seed, n_children=n_patients)
PPGs = []
for i in range(n_patients):
    PPGs.append(nk.ppg_simulate(..., random_state=patients_RNGs[i]))
```